src/super_gradients/training/models/segmentation_models/segmentation_module.py

```diff
@@ -59,3 +70,51 @@ def init_params(self):
             elif isinstance(m, nn.BatchNorm2d):
                 nn.init.constant_(m.weight, 1)
                 nn.init.constant_(m.bias, 0)
+
+    @resolve_param("image_processor", ProcessingFactory())
+    def set_dataset_processing_params(
+        self,
+        class_names: Optional[List[str]] = None,
+        image_processor: Optional[Processing] = None,
+    ) -> None:
+        """Set the processing parameters for the dataset.
+
+        :param class_names:     (Optional) Names of the dataset the model was trained on.
+        :param image_processor: (Optional) Image processing objects to reproduce the dataset preprocessing used for training.
+        """
+        self._class_names = class_names or self._class_names
+        self._image_processor = image_processor or self._image_processor
+
+    @lru_cache(maxsize=1)
+    def _get_pipeline(self, fuse_model: bool = True) -> SegmentationPipeline:
+        """Instantiate the segmentation pipeline of this model.
+        :param fuse_model: If True, create a copy of the model, and fuse some of its layers to increase performance. This increases memory usage.
+        """
+        if None in (self._class_names, self._image_processor):
+            raise RuntimeError(
+                "You must set the dataset processing parameters before calling predict.\n" "Please call `model.set_dataset_processing_params(...)` first."
+            )
+
+        pipeline = SegmentationPipeline(
+            model=self,
+            image_processor=self._image_processor,
+            class_names=self._class_names,
+            fuse_model=fuse_model,
+        )
+        return pipeline
+
+    def predict(self, images: ImageSource, batch_size: int = 32, fuse_model: bool = True) -> ImagesSegmentationPrediction:
```

Comment:
Please add `HasPredict` mixin to inheritance: SegmentationModule(SgModule, ABC, HasPredict):

Reply:
done (4df033eecccf78c3cd10dfe89771242bd3c0abcf)